jsapp/js/components/common/miniAudioPlayer.tsx

```diff
@@ -0,0 +1,218 @@
+    super(props);
+
+    this.state = {
+      isLoading: true,
+      isPlaying: false,
+      currentTime: 0,
+      totalTime: 0,
+    };
+  }
+
+  componentDidUpdate(prevProps: MiniAudioPlayerProps) {
+    if (prevProps.mediaURL !== this.props.mediaURL) {
+      // Reload audio element when URL changes.
+      this.audioInterface.src = this.props.mediaURL;
+    }
+  }
+
+  componentDidMount() {
+    // Prepare audio.
+    this.audioInterface = new Audio(this.props.mediaURL);
+
+    // Set up listeners for audio component.
+    this.audioInterface.addEventListener('loadedmetadata', this.onAudioLoadedBound);
+    this.audioInterface.addEventListener('error', this.onAudioErrorBound);
+    this.audioInterface.addEventListener('timeupdate', this.onAudioTimeUpdatedBound);
+    document.addEventListener(PLAYER_STARTED_EVENT, this.onAnyPlayerStartedBound);
+  }
+
+  componentWillUnmount() {
+    // Pausing makes it subject to garbage collection.
+    this.audioInterface.pause();
+
+    this.audioInterface.removeEventListener('loadedmetadata', this.onAudioLoadedBound);
+    this.audioInterface.removeEventListener('error', this.onAudioErrorBound);
+    this.audioInterface.removeEventListener('timeupdate', this.onAudioTimeUpdatedBound);
+    document.removeEventListener(PLAYER_STARTED_EVENT, this.onAnyPlayerStartedBound);
+  }
+
+  onAnyPlayerStarted(evt: CustomEventInit<string>) {
+    if (this.state.isPlaying && evt.detail !== this.uid) {
+      this.stop();
+    }
+  }
+
+  onAudioError() {
+    this.setState({
+      isLoading: false,
+      isBroken: true,
+    });
+  }
+
+  onAudioLoaded() {
+    this.setState({
+      isLoading: false,
+      isBroken: false,
+      totalTime: this.audioInterface.duration,
+    });
+  }
+
+  onAudioTimeUpdated() {
+    // Pause the player when it reaches the end
+    if (
+      this.audioInterface.currentTime === this.state.totalTime &&
+      this.state.isPlaying
+    ) {
+      this.stop();
+    }
+
+    this.setState({currentTime: this.audioInterface.currentTime});
+  }
+
+  onButtonClick() {
+    if (!this.state.isPlaying) {
+      this.start();
+    } else {
+      this.stop();
+    }
+
+    this.setState({
+      isPlaying: !this.state.isPlaying,
+    });
+  }
+
+  start() {
+    this.audioInterface.play();
+    const event = new CustomEvent(PLAYER_STARTED_EVENT, {detail: this.uid});
+    document.dispatchEvent(event);
+  }
+
+  stop() {
+    // Setting time to 0 and pausing is a silly way to "stop" audio.
+    this.audioInterface.currentTime = 0;
+    this.audioInterface.pause();
+    this.setState({
+      currentTime: 0,
+      isPlaying: false,
+    });
+  }
+
+  renderPlayer() {
+    return (
+      <React.Fragment>
+        <Button
+          type='bare'
+          startIcon={this.state.isPlaying ? 'stop' : 'play'}
+          size='s'
+          color='blue'
+          onClick={this.onButtonClick.bind(this)}
+          data-cy='mini audio player playstop'
```

Comment:
These are identifiers for Cypress tests :) There are being used with `querySelector`. If you see one, make sure that your changes would not break test. I've added the option to define a custom `data-cy` for all common components, and I'm also adding some basic ones for parts of components (to avoid components having multiple properties for cypress).